### PR TITLE
Add API information to README.md

### LINKED ISSUE #30: Add API documentation
Consumers can only find the true capabilities of the generator by looking through examples or searching the interface. The bigger parts like syntax, terminal, background fallbacks etc. should be called out on the readme.

### PR DESCRIPTION
**Pull request by rdnlsmith**
Resolves #30. Here's my first pass; let me know what you think. I'm not 100% sure that my descriptions for the `ui` properties are entirely accurate :grimacing:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +49,77 @@ npm install
...
+
+By default, `comment` is derived from the `background` color, and `modifier` and `markdownQuote` are not set.
+
+The `ui` properties allow you to set colors for various highlights and a few other things:
+
+- `cursor`: the cursor color
+- `invisibles`: used for visible whitespace (see the `editor.renderWhitespace` VS Code setting)
+- `guide`: indentation guidlines in the editor pane
+- `lineHighlight`: colors the line your cursor is on, in place of the boundary lines
+- `findMatchHighlight` and `currentFindMatchHighlight`: higlights matches from the find widget
```
Comment (Tyriar) [inline README.md]: typo: higlights

File: README.md [hunk 2]
```diff
@@ -49,6 +49,77 @@ npm install
...
+- `color3` determines `function`, `class`, `classMember`, `type`, and `cssTag`
+- `color4` determines `functionCall` and `number`
+
+By default, `comment` is derived from the `background` color, and `modifier` and `markdownQuote` are not set.
+
+The `ui` properties allow you to set colors for various highlights and a few other things:
+
+- `cursor`: the cursor color
+- `invisibles`: used for visible whitespace (see the `editor.renderWhitespace` VS Code setting)
+- `guide`: indentation guidlines in the editor pane
```
Comment (Tyriar) [inline README.md]: typo: guidlines

**Comment by rdnlsmith** [comment]
Fixed typos and added the `import` and `generateTheme()` statements to the extended example.

Installed the `streetsidesoftware.code-spell-checker` extension :sweat_smile:

**Comment by Tyriar** [comment]
Thanks @rdnlsmith 😃

**Pull request merged**